test/shopify_app/login_protection_test.rb

```diff
@@ -7,6 +7,7 @@ class LoginProtectionController < ActionController::Base
   helper_method :shop_session
 
   before_action :login_again_if_different_shop, only: [:second_login]
+  around_action :shopify_session, only: [:with_a_shopify_session]
```

Comment:
Neither of the existing controller actions calls `#shopify_session` . I might be able to restructure the test controller and the tests, but that's a lot more work than I'd like to put into this.